### PR TITLE
[App Insights] Adding Data Plane SDK 

### PR DESCRIPTION
**Pull request by alexeldeib**
## Adding Application Insights Data plane Operations 

https://github.com/Azure/azure-rest-api-specs/pull/2404
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/ApplicationInsights/ApplicationInsights.sln [hunk 1]
```diff
@@ -41,6 +45,30 @@ Global
...
 		{4A1E2ACF-218C-481F-A501-7BB3F3C0B873}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{4A1E2ACF-218C-481F-A501-7BB3F3C0B873}.Release|Any CPU.Build.0 = Release|Any CPU
+		{BC9AA77F-1605-4386-B04E-368B1363F395}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{BC9AA77F-1605-4386-B04E-368B1363F395}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{BC9AA77F-1605-4386-B04E-368B1363F395}.Net45-Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{BC9AA77F-1605-4386-B04E-368B1363F395}.Net45-Debug|Any CPU.Build.0 = Debug|Any CPU
+		{BC9AA77F-1605-4386-B04E-368B1363F395}.Net45-Release|Any CPU.ActiveCfg = Release|Any CPU
+		{BC9AA77F-1605-4386-B04E-368B1363F395}.Net45-Release|Any CPU.Build.0 = Release|Any CPU
+		{BC9AA77F-1605-4386-B04E-368B1363F395}.Portable-Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{BC9AA77F-1605-4386-B04E-368B1363F395}.Portable-Debug|Any CPU.Build.0 = Debug|Any CPU
```
Comment (dsgouda) [inline src/SDKs/ApplicationInsights/ApplicationInsights.sln]: What is the intention behind building against portable configurations

**Comment by dsgouda** [comment]
The directory structure needs to be modified here as below 
- Create two directories under `src/SDKs/ApplicationInsights`, `Management` and `DataPlane`
- Move `Management.ApplicationInsights` and `ApplicationInsights.Tests` under `Management`
- Move `Data.ApplicationInsights` and `Data.ApplicationInsights.Tests`  under `DataPlane` 
- Rename `Data.ApplicationInsights` to `ApplicationInsights`

The code has not be generated using `generate.cmd` Please use the script to generate the code, this also generates some additional artifacts required for the CI builds

File: src/SDKs/ApplicationInsights/Data.ApplicationInsights/Microsoft.Azure.ApplicationInsights.csproj [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
```
Comment (dsgouda) [inline src/SDKs/ApplicationInsights/Data.ApplicationInsights/Microsoft.Azure.ApplicationInsights.csproj]: Please add 
```
<!-- Please do not move/edit code below this line -->
	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
<!-- Please do not move/edit code below this line -->
``` 
at the top and 
```
<!-- Please do not move/edit code below this line -->
	<Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.RP.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
<!-- Please do not move/edit code above this line -->
```

at the bottom of this file, as [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/ApplicationInsights/Management.ApplicationInsights/Microsoft.Azure.Management.ApplicationInsights.csproj#L2)

File: src/SDKs/ApplicationInsights/Data.ApplicationInsights/generate.cmd [hunk 3]
```diff
@@ -0,0 +1,7 @@
+::
+:: Microsoft Azure SDK for Net - Generate library code
+:: Copyright (C) Microsoft Corporation. All Rights Reserved.
+::
+
+@echo off
+call %~dp0..\..\..\..\tools\generate.cmd ApplicationInsights/data-plane %*
```
Comment (dsgouda) [inline src/SDKs/ApplicationInsights/Data.ApplicationInsights/generate.cmd]: Please update this call to correspond with the modified directory structure

**Review by dsgouda — CHANGES_REQUESTED** [review]
Requesting some changes.

File: src/SDKs/ApplicationInsights/ApplicationInsights.sln (hunk 1 above)
Comment (alexeldeib) [reply src/SDKs/ApplicationInsights/ApplicationInsights.sln]: None -- not sure when this snuck in. I removed it while changing up the directory structure.

File: src/SDKs/ApplicationInsights/Data.ApplicationInsights/generate.cmd (hunk 3 above)
Comment (alexeldeib) [reply src/SDKs/ApplicationInsights/Data.ApplicationInsights/generate.cmd]: Fixed

**Comment by dsgouda** [comment]
@alexeldeib You probably need to pull latest changes from `psSdkJson6` and run the `generate.cmd` command again

File: tools/generateMetadata.ps1 [hunk 4]
```diff
@@ -6,6 +6,7 @@ Write-Host "GitHub user:" $Args[0]
 Write-Host "Branch:     " $Args[1]
 Try
 {
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
```
Comment (dsgouda) [inline tools/generateMetadata.ps1]: @shahabhijeet 
Checked a couple of blogs, and looks like this is good. Powershell `Invoke-RestMethod` seems to default to the tls 1.0 version which fails with a number of rest calls
An example [here](https://stackoverflow.com/questions/41618766/powershell-invoke-webrequest-fails-with-ssl-tls-secure-channel) 
I couldn't get any specific information regarding whether github also needs later versions of tls but I'm inclined to believe this is a useful change

File: src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.csproj [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
```
Comment (dsgouda) [inline src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.csproj]: Please address previous comments about header and footer for the csproj file

File: tools/generateMetadata.ps1 (hunk 4 above)
Comment (alexeldeib) [reply tools/generateMetadata.ps1]: If you're running agains PS 6.0 or greater, you could also do "Invoke-RestMethod -SslProtocol Tls12" for a slightly cleaner solution, but I don't think the flag exists in earlier versions. Some info on github specifically: https://github.com/blog/1734-improving-our-ssl-setup and scan of their site: https://www.ssllabs.com/ssltest/analyze.html?d=github.com&s=192.30.255.112 
image

**Review by dsgouda — CHANGES_REQUESTED** [review]
csproj needs to be updated per previous comments.

File: tools/generateMetadata.ps1 (hunk 4 above)
Comment (dsgouda) [reply tools/generateMetadata.ps1]: Ahh I missed that!

**Review by dsgouda — APPROVED** [review]
LGTM subject to CI.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**